tapis-project/authenticator
# feat: GH-79: lock fieldsets and buttons on form submit

## Overview

When a form is submitted, disable its fieldsets (thus fields of each fieldset) and buttons.

## Related

- closes #79
- requires https://github.com/TACC/Core-Styles/pull/365

## Changes

- **updated** Core-Styles
- **added** script to disable forms on submit

## Testing

0. Open `/v3/oauth2/webapp`.
1. Submit form.
2. Verify fields become disabled.
3. Verify button becomes disabled.

> [!WARNING]
> I have **not** tested MFA.

## UI

<details open><summary>✅ 3f035e2</summary>

https://github.com/user-attachments/assets/7d48032f-b803-4354-8fb2-b819f9d6946b

✐ When fields are `readonly`, user can select and focus, but **not** type.
✐ When field is not in focus, it has no border.

</details>

<details><summary>❌ c481f78</summary>

⨂ Uses `disabled` which prevents submission of data.

| login | mfa |
| - | - |
| <img width="1194" alt="GH-79 login" src="https://github.com/user-attachments/assets/4ad5be95-24ca-4ba9-92f6-2dcedcdd07ae"> | ❓ |

https://github.com/user-attachments/assets/f2c19f79-570a-48f6-b58c-225a8977bd03

</details>
